pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -133,6 +135,38 @@ func validateContainer(container corev1.Container) *apis.FieldError {
 	return errs
 }
 
+func validateBuildRef(buildRef *corev1.ObjectReference) *apis.FieldError {
+	if buildRef == nil {
+		return nil
+	}
+	if len(validation.IsQualifiedName(buildRef.APIVersion)) != 0 {
+		return apis.ErrInvalidValue(buildRef.APIVersion, "apiVersion")
+	}
+	if len(validation.IsCIdentifier(buildRef.Kind)) != 0 {
+		return apis.ErrInvalidValue(buildRef.Kind, "kind")
+	}
+	if len(validation.IsDNS1123Label(buildRef.Namespace)) != 0 {
```

Comment:
We should disallow namespace.

Reply:
I have no problem disallowing it, but would it be okay if `buildRef.namespace` has the same value as `revision.namespace`?